lib/core/test/integration/Test/Integration/Scenario/CLI/Server.hs

```diff
@@ -4,67 +4,51 @@ module Test.Integration.Scenario.CLI.Server
-            removeDirectory d
-            launcher d
-            ls <- listDirectory d
-            ls `shouldContain` ["wallet.db"]
-
-    describe "DaedalusIPC" $ do
-        it "should reply with the port when asked" $ do
-            (_, _, _, ph) <-
-                createProcess (proc "test/integration/js/mock-daedalus.js" [])
-            waitForProcess ph `shouldReturn` ExitSuccess
-
-withTempDir :: (FilePath -> IO a) -> IO a
-withTempDir = withSystemTempDirectory "integration-state"
-
-launcher :: FilePath -> IO ()
-launcher stateDir = withCreateProcess cmd $ \_ _ (Just stderr) ph -> do
-    TIO.hGetContents stderr >>= TIO.putStrLn
-    terminateProcess ph
-  where
-    cmd = proc' "cardano-wallet" ["launch", "--state-dir", stateDir]
-
--- There is a dependency cycle in the packages.
---
--- cardano-wallet-launcher depends on cardano-wallet-http-bridge so that it can
--- import the HttpBridge module.
---
--- This package (cardano-wallet-http-bridge) should have
--- build-tool-depends: cardano-wallet:cardano-wallet-launcher so that it can
--- run launcher in the tests. But that dependency can't be expressed in the
--- cabal file, because otherwise there would be a cycle.
---
--- So one hacky way to work around it is by running programs under "stack exec".
-proc' :: FilePath -> [String] -> CreateProcess
-proc' cmd args = (proc "stack" (["exec", "--", cmd] ++ args))
-    { std_in = CreatePipe, std_out = CreatePipe, std_err = CreatePipe }
+    ( SpecWith, after_, describe, it )
+import Test.Hspec.Expectations.Lifted
+    ( shouldBe )
+import Test.Integration.Framework.DSL
+    ( Context (..), expectCmdStarts )
+
+spec :: SpecWith (Context t)
+spec = after_ tearDown $ do
+    describe "SERVER - cardano-wallet serve" $ do
+        it "SERVER - Can start cardano-wallet serve without --database" $ \_ -> do
+            let cardanoWalletServer = Command "stack"
+                    [ "exec", "--", "cardano-wallet", "serve"
+                    ] (return ())
+                    Inherit
+            expectCmdStarts cardanoWalletServer
+
+            w1 <- doesFileExist dbFile
+            w2 <- doesFileExist (dbFile ++ "-shm")
+            w3 <- doesFileExist (dbFile ++ "-wal")
+            w1 `shouldBe` False
+            w2 `shouldBe` False
+            w3 `shouldBe` False
+
+        it "SERVER - Can start cardano-wallet serve with --database" $ \_ -> do
+            let cardanoWalletServer = Command "stack"
+                    [ "exec", "--", "cardano-wallet", "serve"
+                    , "--database", dbFile
+                    ] (return ())
+                    Inherit
+            expectCmdStarts cardanoWalletServer
+
+            w1 <- doesFileExist dbFile
+            w2 <- doesFileExist (dbFile ++ "-shm")
+            w3 <- doesFileExist (dbFile ++ "-wal")
+            w1 `shouldBe` True
+            w2 `shouldBe` True
+            w3 `shouldBe` True
+ where
+     dbFile = "./test/data/test-DB-File"
```

Comment:
@rvl's version was using temporary random directory which has the nice side effect of not cluttering the development environment and avoid file conflicts between multiple runs (because for some reasons / failures, the tear down code wouldn't run).

Reply:
>(because for some reasons / failures, the tear down code wouldn't run).

I suppose `after_ tearDown` should run no matter if test fails or passes?

Also @rvl's tests where for the `launch` and `--state-dir` not for `serve` and `--database` (https://github.com/input-output-hk/cardano-wallet/blob/master/lib/core/test/integration/Test/Integration/Scenario/CLI/Server.hs#L28-L39)

In any case I can bring temporary dir back.

Reply:
> I suppose after_ tearDown should run no matter if test fails or passes?

Provided the tear down code works as expected and removes files where they ought to be :grimacing: ... Perhaps files are already there and are created elsewhere, so the test doesn't test anything ?  The good thing with taking a random file location everytime is that, we are "sure" nothing was there before, no matter what and we can eliminate false positives. 

Although, that should also hold for the CI environment (which is supposedly, free of any build and tests artifacts), but it's not like we've never mistakenly committed test files and had them available in the integration environment :roll_eyes: ...

Reply:
ok, added temp directory for `launch` and `serve` tests